ui-admin/src/portal/DataImportView.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState } from 'react'
+import Api, { DataImportItem } from 'api/api'
+import LoadingSpinner from 'util/LoadingSpinner'
+import { ColumnDef, getCoreRowModel, getSortedRowModel, SortingState, useReactTable } from '@tanstack/react-table'
+import {
+  basicTableLayout,
+  DownloadControl,
+  IndeterminateCheckbox,
+  renderEmptyMessage,
+  RowVisibilityCount
+} from 'util/tableUtils'
+import { currentIsoDate, instantToDefaultString } from '@juniper/ui-core'
+import { useLoadingEffect } from '../api/api-utils'
+import { renderPageHeader } from 'util/pageUtils'
+import { StudyEnvContextT, useStudyEnvParamsFromPath } from '../study/StudyEnvironmentRouter'
+import { useParams } from 'react-router-dom'
+
+
+/** show the dataImportItem list in table */
+export default function DataImportView({ studyEnvContext }:
+                                           { studyEnvContext: StudyEnvContextT }) {
+  const [dataImportItems, setDataImportItems] = useState<DataImportItem[]>([])
+  const [sorting, setSorting] = React.useState<SortingState>([{ 'id': 'createdAt', 'desc': true }])
+  const [rowSelection, setRowSelection] = React.useState<Record<string, boolean>>({})
+  const columns: ColumnDef<DataImportItem>[] = [{
+    id: 'select',
+    header: ({ table }) => <IndeterminateCheckbox
+      checked={table.getIsAllRowsSelected()} indeterminate={table.getIsSomeRowsSelected()}
+      onChange={table.getToggleAllRowsSelectedHandler()}/>,
+    cell: ({ row }) => (
+      <div className="px-1">
+        <IndeterminateCheckbox
+          checked={row.getIsSelected()} indeterminate={row.getIsSomeSelected()}
+          onChange={row.getToggleSelectedHandler()} disabled={!row.getCanSelect()}/>
+      </div>
+    )
+  },
+  {
+    header: 'EnrolleeId',
+    accessorKey: 'createdEnrolleeId'
+  },
+  {
+    header: 'Imported Date',
+    accessorKey: 'createdAt',
+    meta: {
+      columnType: 'instant'
+    },
+    cell: info => instantToDefaultString(info.getValue() as number)
+  },
+  {
+    header: 'Status',
+    accessorKey: 'status'
+  },
+  {
+    header: 'Message',
+    accessorKey: 'message'
+  }]
+
+  const table = useReactTable({
+    data: dataImportItems,
+    columns,
+    state: {
+      sorting,
+      rowSelection
+    },
+    enableRowSelection: true,
+    onSortingChange: setSorting,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    onRowSelectionChange: setRowSelection
+  })
+
+  const { dataImportId } = useParams()
+  if (!dataImportId) {
+    return <></>
+  }
+  //const numSelected = Object.keys(rowSelection).length
+  const studyEnvParams = useStudyEnvParamsFromPath()
+  const studyShortCode = studyEnvParams.studyShortcode
+  if (!studyShortCode) {
+    return <></>
+  }
+
+  const { isLoading } = useLoadingEffect(async () => {
+    const result = await Api.fetchDataImport(studyEnvContext.portal.shortcode, studyShortCode,
```

Comment:
DataImportList already has this data. Maybe I could have passed it rather than additional call !

Reply:
sure, but I like refetching here.  this isn't performance sensitive code, so I'd rather be sure we're getting the latest data from the server